compiler/typecheck/TcRnTypes.hs

```diff
@@ -3827,6 +3829,8 @@ pprCtO AnnOrigin             = text "an annotation"
 pprCtO HoleOrigin            = text "a use of" <+> quotes (text "_")
 pprCtO ListOrigin            = text "an overloaded list"
 pprCtO StaticOrigin          = text "a static form"
+pprCtO NonLinearPatternOrigin = text "a non-linear pattern"
```

Comment:
Is it non-linear, or non-unrestricted?

Reply:
I'm calling patterns such as `_` non-linear, possibly we could call them unrestricted.